src/ray/stats/metric_exporter.cc

```diff
@@ -25,10 +25,21 @@ void MetricExporter::ExportToPoints(
         &view_data,
     const std::string &metric_name, std::vector<std::string> &keys,
     std::vector<MetricPoint> &points) {
+  RAY_CHECK(is_exporting_ == false);
   // Return if no raw data found in view map.
   if (view_data.size() == 0) {
     return;
   }
+
+  if (should_be_cleaned_up_) {
```

Comment:
This means once cleanup is called, we won't export any metrics. It is sort of a hacky way to avoid segfault that can happen in the termination process.